thoth/storages/graph/postgres.py

```diff
@@ -4822,6 +4894,57 @@ def sync_revsolver_result(self, document: Dict[str, Any]) -> None:
                     platform=entry["platform"],
                 )
 
+    def _check_package_solved(self, session: Session, package_name: str, package_version: str,
+                              package_index: str, os_name: str, os_version: str, python_version: str) -> Tuple:
+        """We check if the package has been solved before syncing it to the database."""
+        # Check if the package has a known index
+        if package_index:
+            python_package_version = (
+                session.query(PythonPackageVersion)
+                .join(PythonPackageIndex)
+                .filter(PythonPackageVersion.package_name == package_name)
+                .filter(PythonPackageVersion.package_version == package_version)
+                .filter(PythonPackageVersion.os_name == os_name)
+                .filter(PythonPackageVersion.os_version == os_version)
+                .filter(PythonPackageVersion.python_version == python_version)
```

Comment:
This is not required also :) See comment above

Reply:
This should be substituted  with just a check if package_version, version, index and return the record otherwise fail.

Reply:
Thanks I have removed it